tests/post/steps/test_seccomp.py

```diff
@@ -0,0 +1,59 @@
+import os.path
+
+import yaml
+
+from kubernetes import client
+
+import pytest
+from pytest_bdd import scenario, when
+
+from tests import kube_utils
+from tests import utils
+
+
+@scenario("../features/seccomp.feature",
+          "Running a Pod with the 'runtime/default' seccomp profile works")
+def test_seccomp(host):
+    pass
+
+
+@when("we create a utils Pod with labels {'test': 'seccomp1'} "
+      "and annotations "
+      "{'seccomp.security.alpha.kubernetes.io/pod': 'runtime/default'}")
+def create_utils_pod(utils_pod):
+    pass
+
+
+@pytest.fixture
+def utils_pod(k8s_client, utils_image):
```

Comment:
You don't need to create another fixture, `@when` already makes the `create_utils_pod` a fixture. This would allow us to make use of a BDD parser:

```python
_UTILS_POD_PARSER = parsers.parse(
    "we create a utils Pod with labels {labels:JSON} "
    "and annotations {annotations:JSON}",
    extra_types=dict(JSON=json.loads),
)

@when(_UTILS_POD_PARSER)
def utils_pod(k8s_client, utils_image, labels, annotations):
    # create the Pod with right labels and annotations, which are
    # passed in as Python dicts already
    try:
        yield pod_name
    finally:
        # clean-up of the Pod

@then("I can run '{command}' in the utils Pod")
def run_in_utils_pod(utils_pod, command):
     # If we had such a helper, we could do:
    kubectl_exec(pod_name=utils_pod, command=command)
```

Reply:
Well, that's what I tried at first (how could one perform resource cleanup otherwise?). However, when executing the test using that implementation, nothing happened. My `when` thing, which is then a generator, would not be `next`ed, so no code executed, no Pod created,... (to my surprise). Skimming the docs back then, there's no reference of `when`s being permitted to be a generator.

I'll try again with a super-simple test...

Reply:
As discussed on Slack, one can't use a generator in a `@when` implementation.

Reply:
Indeed, but it seems we can in a `@given` (and in your context, creating a utils Pod could make sense as a "Given" step).

Reply:
Well, no :) I am not actually *using* the `utils` Pod, its merely a canary (hence, not executing any commands in it). What this test wants to validate is that *any* Pod with some specific set of annotations will be fully functional at some point (i.e., our CRI supports the `seccomp` profile annotations). The Pod is not an input to the test, its creation and validation *is* the test.